src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -932,4 +1060,205 @@ public FormValidation doCheckCreateSnapshot(
       return FormValidation.ok();
     }
   }
+
+  // For use in Builder only
+  private InstanceConfiguration() {}
+
+  public static class Builder {
+
+    private InstanceConfiguration instanceConfiguration;
+
+    public Builder() {
+      instanceConfiguration = new InstanceConfiguration();
+    }
+
+    public Builder description(String description) {
+      instanceConfiguration.description = description;
```

Comment:
The builder should use the setters rather than breaking encapsulation. This allows for precondition checks and logging as needed.

Reply:
Will do